gnosis/gnosis-management
# Feature: Live Refreshing Wallet

# Description:

### `WalletIntegrationProvider` Refactor
I've improved the functionality of the wallet provider layer and implemented a 'watcher' system to all integrations. Currently it's not supported to listen to any kind of event on a web3 Provider so we're instead updating every second, if something about the integration has changed. (Currently: Network, Account or Balance)

### Provider Priority System
Previously, if all other wallet providers failed to connect, the `REMOTE` provider would be used instead. I've now added a var called `providerPriority` to all integration, which enables us to selectively prefer integrations over others. (For example prefer to use Parity even if Metamask is available because Parity is a dedicated browser)

### Current Provider Information in Redux Store
[See here for quick demo](https://i.imgur.com/54QFx1Z.gifv)

The currentProvider is now a key in the blockchain redux state. The selectors have been changed so only the currentProvider Information is displayed. Inside the Header we now display the Network if it's not the main network. Provider Icon Alt-Text is the Providers Name if the Icon is not clear. Identicon Alt-Text is Account Address.

#### Changed functionality:
- Modal is now only visible if no connection could be made to any sort of wallet. So either connectivity problems or blocked requests. (I know we'll alter this anyway according to new specs)

# Testing:
- Change your Metamask User and watch the Identicon Update
- Change your Network or Account and watch the Balance change
- Change your Provider or disable your current one and see the Icon update
- Lock your Metamask and see that the current Information will disappear
- Inside the Redux State the currentProvider should be set after all providers have (tried) to initialized
